### PR TITLE
LM-178 Added logic for cache-dir management

### PR DESCRIPTION
**Pull request by dusktreader**

### CODE REVIEW

File: src/templates/license-manager.defaults.template [hunk 1]
```diff
@@ -15,3 +15,4 @@ LM2_AGENT_AUTH0_DOMAIN={{ auth0_domain }}
 LM2_AGENT_AUTH0_AUDIENCE={{ auth0_audience }}
 LM2_AGENT_AUTH0_CLIENT_ID={{ auth0_client_id }}
 LM2_AGENT_AUTH0_CLIENT_SECRET={{ auth0_client_secret }}
+LM2_AGENT_CACHE_DIR=/var/cache/license-manager
```
Comment (julianaklulo) [inline src/templates/license-manager.defaults.template:18]: Why is this hard coded?
Isn't the idea to use a env var so we can use a different path when running in other environments?

File: src/license_manager_agent_ops.py [hunk 2]
```diff
@@ -41,6 +43,14 @@ def _derived_pypi_url(self):
         return (f"https://{pypi_username}:{pypi_password}@"
                 f"{url}/simple/{self._PACKAGE_NAME}")
 
+    def setup_cache_dir(self):
+        cache_dir = Path("/var/cache/license-manager")
```
Comment (julianaklulo) [inline src/license_manager_agent_ops.py:47]: We won't be using an env var to set this?

Comment (heitorPB) [reply src/license_manager_agent_ops.py:47]: Why not make it a configurable?

File: src/license_manager_agent_ops.py [hunk 3]
```diff
@@ -41,6 +43,14 @@ def _derived_pypi_url(self):
...
+        if not cache_dir.exists():
+            cache_dir.mkdir()
+            shutil.chown(cache_dir, user="slurm")
+            cache_dir.chmod(0o700)
```
Comment (heitorPB) [inline src/license_manager_agent_ops.py:51]: Why the directory should have executable bit set? Isn't `-rw-------` enough?

Comment (dusktreader) [reply src/license_manager_agent_ops.py:51]: execute bit has to be set to execute `ls` and `cd` in it.

File: src/license_manager_agent_ops.py (hunk 2 above)
Comment (dusktreader) [reply src/license_manager_agent_ops.py:47]: James and I discussed this yesterday. We decided making it hard-coded in the charm removes the overhead of specifying some path that has to be validated and have permissions adjusted and such. We can instead, with the charm, just have a fixed place where the cache lives.

Do you see an advantage to having it configurable?

Comment (heitorPB) [reply src/license_manager_agent_ops.py:47]: Making it configurable has the advantage that we can change the location on disk easily. But we don't need that for now. @julianaklulo I'm fine with the code this way, do you see any problems that we might have here?

Comment (julianaklulo) [reply src/license_manager_agent_ops.py:47]: I agree with you @heitorPB, we can keep it this way for now.

**Review by heitorPB — APPROVED** [review]

**Review by julianaklulo — APPROVED** [review]

**Pull request merged**